### ISSUE TITLE
Installation and running shortest path notebook comments

### ISSUE THREAD

**Issue by akazachk** [khalil-research/PyEPO#2]
I encountered some issues (now resolved) when running the Shortest Path notebook on my laptop running MacOS Monterey 12.2.1 with Python 3.9.10 installed via Homebrew.

After installation, I went to the shortest path notebook.
* I had an issue with `plt.style.use("science")` --- `SciencePlots` is a dependency, but I did not see it listed, so I had to install that.
* I also recommend making the comments in the file a bit more detailed. Examples:
    - In the "Init Data" section, it may help to add more explanation of the parameters for [genData](https://khalil-research.github.io/PyEPO/build/html/content/examples/data.html#pyepo.data.shortestpath.genData) in the cell comments, such as that costs are calculated using `deg`. Or to simply link to the documentation.
    - When calling [optdataset](https://khalil-research.github.io/PyEPO/build/html/content/examples/data.html#optdataset), the current comments can be more informative (e.g., "get training data set, by computing optimal solution and optimal values for each sample cost vector in `c_train`", or whatever else is done there --- are the features used?).
    - After training the SPO+ loss, there are two plots, and both are commented as `# draw plot`. It took a moment to identify what the second plot is showing.
* I think the title + axes need to be relabeled in the second plot, if I understood correctly.
* When comparing solutions after training for the SPO+ loss, it may help to plot the values of the pred vs actual cost, and pred vs actual solution, to have a visual for what is now currently given as text. Maybe just for the first sample, or maybe there is a nice way to aggregate the data for all the samples.
* Just like the small example in the SPO paper (with 1-d features), it would be good to have a direct comparison of predictions with linear regression vs SPO+. Same for decisions.

A final comment: the documentation is great and helped a lot when I had questions! Thank you.

**Comment by LucasBoTang** [comment]
Hi Aleksandr! Thank you for your helpful suggestion!

- I removed the SciencePlots requirement
- I updated the new [notebook](https://github.com/khalil-research/PyEPO/blob/main/notebooks/01%20Modeling%20and%20Training.ipynb)
- The y-axis of plots  was relabeled
- Plotting the values of the pred vs. actual cost/solution is a good idea. I will add this visualization later.
- For more experiments and figures in the paper, you can see the notebook in the develpment branch.

**Comment by LucasBoTang** [comment]
Hi Aleksandr! I upload [./notebooks/02 2D knapsack Visualization.ipynb](https://github.com/khalil-research/PyEPO/blob/main/notebooks/02%202D%20knapsack%20Visualization.ipynb), which try to visualize and compare true solution and predicted solution.

**Comment by LucasBoTang** [comment]
Hi Aleksandr! New notebooks are in. Hope it will be more clear and helpful :)

- [01 Optimization Model](https://github.com/khalil-research/PyEPO/blob/main/notebooks/01%20Optimization%20Model.ipynb): Build up optimization solver
- [02 Optimization Dataset](https://github.com/khalil-research/PyEPO/blob/main/notebooks/02%20Optimization%20Dataset.ipynb): Generate synthetic data and use optDataset 
- [03 Training and Testing](https://github.com/khalil-research/PyEPO/blob/main/notebooks/03%20Training%20and%20Testing.ipynb): Train and test end-to-end predict-then-optimize for different approaches
- [04 2D knapsack Solution Visualization](https://github.com/khalil-research/PyEPO/blob/main/notebooks/04%202D%20knapsack%20Solution%20Visualization.ipynb): Visualize solutions for knapsack problem
- [05 Warcraft Shortest Path](https://github.com/khalil-research/PyEPO/blob/main/notebooks/05%20Warcraft%20Shortest%20Path.ipynb): Use the Warcraft terrains dateset to train shortest path
